diff --git a/Makefile b/Makefile
index a2b15e5..96f8424 100644
--- a/Makefile
+++ b/Makefile
@@ -188,8 +188,8 @@ else
 # Emscripten-specific build options
 ifeq ($(OS),emscripten)
 override CFLAGS += -pthread
-override LDFLAGS += -s TOTAL_MEMORY=512MB
-override BIN_EXT := .html
+override LDFLAGS += -s TOTAL_MEMORY=96MB
+override BIN_EXT := .mjs
 override LIB_EXT := .so
 USE_SDL ?= 2
 USE_NET ?= 0
